cmake/llext-edk.cmake

```diff
@@ -101,11 +101,11 @@ foreach(flag ${llext_cflags})
         relative_dir(${parent} dest bindir)
         cmake_path(RELATIVE_PATH dest BASE_DIRECTORY ${llext_edk} OUTPUT_VARIABLE dest_rel)
         if(bindir)
-            list(APPEND imacros_gen_make "-imacros\$(${install_dir_var})/${dest_rel}/${name}")
-            list(APPEND imacros_gen_cmake "-imacros\${CMAKE_CURRENT_LIST_DIR}/${dest_rel}/${name}")
+            list(APPEND imacros_gen_make "-imacros \$(${install_dir_var})/${dest_rel}/${name}")
+            list(APPEND imacros_gen_cmake "-imacros \${CMAKE_CURRENT_LIST_DIR}/${dest_rel}/${name}")
         else()
-            list(APPEND imacros_make "-imacros\$(${install_dir_var})/${dest_rel}/${name}")
-            list(APPEND imacros_cmake "-imacros\${CMAKE_CURRENT_LIST_DIR}/${dest_rel}/${name}")
+            list(APPEND imacros_make "-imacros \$(${install_dir_var})/${dest_rel}/${name}")
+            list(APPEND imacros_cmake "-imacros \${CMAKE_CURRENT_LIST_DIR}/${dest_rel}/${name}")
```

Comment:
be careful on CMake de-duplication of symbols.

If there are two `-imacros <foo> -imacros <bar>`, then CMake will perform de-duplication so you can end up with `-imacros <foo> <bar>` unless care is taken.

See more on de-duplication: https://cmake.org/cmake/help/latest/command/add_compile_options.html#option-de-duplication

Note: this comment is purely based on the changed lines seen in this commit, there might be other part of the code which ensures there can only be a single `-imacros` passed on.

Reply:
That's funny - because the generated flags is used in a `add_compile_options` call, but I didn't see the de-duplication taking place. Anyway, this patch was more of a nice to have than a must (for now, at least), so I simply dropped it.

Reply:
> because the generated flags are used in a add_compile_options call, but I didn't see the de-duplication taking place. 

it might depends on exactly how the list is actually being used, hence the reason I just wrote "be careful on CMake  de-duplication of symbols.".

But not changing the line is a sure bet.